tips/TIP-0042/tip-0042.md

```diff
@@ -0,0 +1,1230 @@
+creation, so they are disallowed:
+
+- Storage Deposit Return: Having to return the funds on an implicit account to the sender means the full account that
+  needs to be transitioned to, must be funded by other means. In a typical onboarding scenario these funds do not exist,
+  so SDRUC would only make it more complicated to create the full account.
+- Timelock: Would be harmless, but timelocks are not permitted on full accounts either. It would be odd to be able to
+  use the implicit account to issue blocks (because the accounts ledger does not respect the timelock) but not be able
+  to do the implicit account conversion until the timelock expires.
+- Expiration: To fulfil syntactic rules, the `Return Address` would not be an implicit account creation address, but
+  some other regular address. Then for the `Address`, the Basic Output would be an implicit account, but for the
+  `Return Address` it would be a regular output, which is very undesirable behavior, since the `Return Address` would
+  not have to respect the account-locked mana rules, which would open an attack vector.
+
+## Expiry Slot and End Epoch
+
+One recurring pattern that applies to Staking and Block Issuer Features is the use of the maximum value of slot or epoch
+indices. The max value can be used to express that such features do not have an expiration set, applying to the
+`End Epoch` and `Expiry Slot`, respectively.
+
+Typically, one would want to set these features to not expire and only reduce the value to the closest possible one,
+once removal of the feature is desired.
+
+# Test Vectors
+
+The protocol parameters used in the following test vectors are the same as in
+[TIP-49 (Protocol Parameters Hash)](../TIP-0049/tip-0049.md#protocol-parameters-hash).
+
+## Storage Score
+
+The following test vector shows the calculation of the storage score according to [TIP-47](../TIP-0047/tip-0047.md).
+
+Account Output (json-encoded):
+
+```json
+{
+  "type": 1,
+  "amount": "200000000",
+  "mana": "333000000",
+  "accountId": "0xe8494fe353f99783d3771c78798e1e839e649310513770fc6dc974fe53cf1a86",
+  "foundryCounter": 0,
+  "unlockConditions": [
+    {
+      "type": 0,
+      "address": {
+        "type": 0,
+        "pubKeyHash": "0xed1484f4d1f7d8c037087fed661dd92faccae1eed3c01182d6fdd6828cea144a"
+      }
+    }
+  ],
+  "features": [
+    {
+      "type": 6,
+      "expirySlot": 888,
+      "blockIssuerKeys": [
+        {
+          "type": 0,
+          "publicKey": "0x9e05a32eafedefd40298e24ad4f8c334580187f7e9afbd9da13b5ba4007dd1b5"
+        },
+        {
+          "type": 0,
+          "publicKey": "0xa504844f7a0df2c5101d31696593b309040f8660d41035aba508f24c00668b21"
+        }
+      ]
+    },
+    {
+      "type": 7,
+      "stakedAmount": "150000000",
+      "fixedCost": "400",
+      "startEpoch": 25,
+      "endEpoch": 4294967295
+    }
+  ],
+  "immutableFeatures": [
+    {
+      "type": 0,
```

Comment:
Sender features cannot be immutable features?

Reply:
You're absolutely right, not sure how that ended up there. Updated the test vector. Thanks!

Reply:
Moving the feature changes the serialized bytes FYI

Reply:
I regenerated the test vector and the binary format was updated.

Reply:
Ah, missed that. Thanks!